### PR TITLE
pybrain added to python packages

### PR DESCRIPTION
**Pull request by NikolaMandic**
###### Things done
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x ] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x ] Tested execution of all binary files (usually in `./result/bin/`)
- [x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

tested with  `nix-shell --pure` and then trying it out

---

### CODE REVIEW

**Comment by joachifm** [comment]
Please review the contribution guidelines.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -26399,6 +26399,26 @@ in modules // {
     };
   };
 
+  pybrain = buildPythonPackage rec {
+    name = "pybrain-${version}";
+    version = "0.3.3";
+    src = pkgs.fetchgit {
```
Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: `fetchFromGitHub` is a more convenient way to do this

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -26399,6 +26399,26 @@ in modules // {
...
+      url = https://github.com/pybrain/pybrain.git;
+      rev = "refs/tags/${version}";
+      sha256 = "1i1nqba66dr45sir838l5wy93dim81kxszlrcw96ij5p7s180c81";
+    };
+
+    propagatedBuildInputs = with self; [ scipy ];
+
+    meta = {
```
Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: I suggest adding yourself as the maintainer of this package

**Comment by mbakke** [comment]
Please use packages from PyPi when available: https://pypi.python.org/pypi/PyBrain

**Comment by NikolaMandic** [comment]
thanks guys hopefully I will add more

**Comment by NikolaMandic** [comment]
@mbakke it is available there but is more trouble to install since it wants libraries etc. what do you say is it better to just grab it from github or to find those libs?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -26399,6 +26399,26 @@ in modules // {
...
+      homepage = "http://pybrain.org/";
+      description = "PyBrain is a modular Machine Learning Library for Python.";
+      license = licenses.bsd;
+    };
+
+    doCheck = false; # lazy packager
+  };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: do you mean no tests are included? Then please right that.

Comment (NikolaMandic) [inline pkgs/top-level/python-packages.nix]: how do you include test for this please tell me. I've tested manually

**Comment by NikolaMandic** [comment]
can people review it again @mbakke would you let it be without pypy if it is easier or do you say go for extra length because pypy is worth it?

**Comment by joachifm** [comment]
What do you mean by the pypi version needing libraries whereas the github one does not? Are they bundled in the github version?

**Comment by FRidh** [comment]
@NikolaMandic @mbakke PyPI is indeed the preferred source for packages. 
Unfortunately no packages are made available via https://pypi.python.org/pypi/PyBrain/0.3.3

In that case using GitHub or another source is fine, but do please try to use archives/tarballs instead of repositories. You can find archives of the releases on this page
https://github.com/pybrain/pybrain/tags
and you need to use specifically this tarball
https://github.com/pybrain/pybrain/archive/0.3.3.tar.gz

**Comment by NikolaMandic** [comment]
@joachifm when I do it with pypy it does download thing from some sources link but it can not find libraries for dependencies while setup.py install no problem.

@FRidh  I will do it please tell me why is it preferred

**Comment by joachifm** [comment]
@NikolaMandic it's preferred in the same way hackage is preferred for Haskell packages or CPAN for Perl. Relying on autogenerated tarballs from github isn't great. Anyway, it's okay to use github if upstream does not publish up-to-date packages on pypi.

**Comment by NikolaMandic** [comment]
@FRidh again library issue if from tarball
the thing is setup py takes care of dependencies while other way I would have to download libs make it available for the linker etc. it looks like yak shave

**Comment by NikolaMandic** [comment]
well in this case it will clone the thing and then install with setup py. it wont take any tarballs

**Comment by FRidh** [comment]
@NikolaMandic What libraries are you talking about? 
`setup.py` lists only `scipy` as a dependency, which you correctly added to `propagatedBuildInputs`. 
And you can use an archive as source by using fetchurl.
```
src = pkgs.fetchurl {
  url = "https://github.com/pybrain/pybrain/archive/${version}.tar.gz";
  sha256 = "<fill it in>";
};
```

**Comment by NikolaMandic** [comment]
last issue with your thing was libdiscid.so.0 and before it wanted two libraries if I remember correctly it wanted blas and some other thing

**Comment by NikolaMandic** [comment]
@joachifm @FRidh ping just in case

**Comment by NikolaMandic** [comment]
closed by github for some reason while I rebased and force pushed

**Comment by NikolaMandic** [comment]
@joachifm @FRidh  I've changed this to get archive instead to clone

**Comment by FRidh** [comment]
Thanks. In the future please prefix the package name with `pythonPackages.` as I noted in your other PR.

**Pull request merged**